test/integration/ads_integration_test.cc

```diff
@@ -434,5 +435,58 @@ TEST_P(AdsFailIntegrationTest, ConnectDisconnect) {
   ads_stream_->finishGrpcStream(Grpc::Status::Internal);
 }
 
+class AdsConfigIntegrationTest : public HttpIntegrationTest,
```

Comment:
I'm wondering if this could have been also done as a server unit test, i.e. just verifying the config loads. I think it's fine to do this here, but I'd also try and minify the amount of stuff we have; all we want to do is verify initialization with an ADS configuration. We don't care about setting up an ADS server or actually serving up ADS.

Reply:
i agree. I thought about it being a unit test. I added it here so that we could add ADS related config tests that ensure ADS connection works with config changes in future if needed.